airbyte-cdk/python/airbyte_cdk/sources/file_based/availability_strategy/default_file_based_availability_strategy.py

```diff
@@ -64,6 +64,8 @@ def check_availability_and_parsability(
     def _check_list_files(self, stream: "AbstractFileBasedStream") -> List[RemoteFile]:
         try:
             files = stream.list_files()
+        except CustomFileBasedSourceException as custom_exc:
```

Comment:
@tolik0 thanks for this PR, I agree that we want to be able customize the exceptions.

Looking at this code: the error that's relevant to the user is still that there was a problem listing files, so I think we want to keep the `CheckAvailabilityError`, but want to make sure we can customize the message, e.g. for things like missing bucket.

Since we're already raising `CheckAvailabilityError ... from exc`, we should get the missing bucket error as part of the stacktrace when we raise the `CheckAvailabilityError`, right?

If that's obscuring the error too much, instead of raising  `CustomFileBasedSourceException`, WDYT about giving it a "message" attribute, then here we'll raise `CheckAvailabilityError(custom_exc.message, stream=stream.name) from exc`?

Reply:
@clnoll Thanks for feedback!
I've removed **CustomFileBasedSourceException**. Do you think this is better? Should I split this PR into two: one for file-based CDK and another for S3 changes?